src/app_service/service/accounts/service.nim

```diff
@@ -295,14 +295,15 @@ QtObject:
       "profile-pictures-show-to": settings.PROFILE_PICTURES_SHOW_TO_EVERYONE,
       "profile-pictures-visibility": settings.PROFILE_PICTURES_VISIBILITY_EVERYONE,
       "url-unfurling-mode": int(settings.UrlUnfurlingMode.AlwaysAsk),
+      "omit-transfers-history-scan?": omitTransfersHistory,
     }
 
   proc getAccountSettings(self: Service, accountId: string,
     installationId: string,
     displayName: string): JsonNode =
     for acc in self.generatedAccounts:
       if(acc.id == accountId):
-        return self.prepareAccountSettingsJsonObject(acc, installationId, displayName)
+        return self.prepareAccountSettingsJsonObject(acc, installationId, displayName, true)
```

Comment:
@saledjenic, to clarify: we need to pass and check `recoverAccount` because one of generated accounts might be recovered or something, or there is another reason?

Reply:
You're setting `omitTransfersHistory` to `true`, but it should not be set to `true` in case the user is recovering, so from the `setupAccount` function, you have to pass `false` to `prepareAccountSettingsJsonObject` over `getAccountSettings` for `omitTransfersHistory` param.

Reply:
@rasom but I am still wondering why we need that?!?!
Why don't we use `MnemonicRemoved` from the settings and perform a history check only if `MnemonicRemoved` is set to `false`? That's the cleanest way and the most correct one.

In the desktop app user can create a new Status account following any of the following onboarding paths:
1. NewUser -> `Generate new keys`
2. NewUser -> `Generate keys for a new Keycard`
3. NewUser -> `lmport a seed phrase` -> `lmport a seed phrase`
4. NewUser -> `lmport a seed phrase` -> `lmport a seed phrase into a new Keycard`
5. OldUser -> `Scan sync code`
6. OldUser -> `Login with Keycard`
7. OldUser -> `Enter a seed phrase`

Now only in case 1, `MnemonicRemoved` will be set to `false`, in all other cases it's `true`. 
Why don't we use that info on the status-go side, but instead we're setting explicitly `"omit-transfers-history-scan?"`?

Reply:
> You're setting omitTransfersHistory to true, but it should not be set to true in case the user is recovering

but that's iteration over `self.generatedAccounts`, recovered accs are also there?

Reply:
> but that's iteration over self.generatedAccounts, recovered accs are also there?

Yes, that's some legacy, but anyhow will be set for a single account only, not for all. We're integrating to find the correct one and proceed with it.

Reply:
> but I am still wondering why we need that?!?!

we don't, and no change on the client side is needed for `CreateAccountAndLogin/RestoreAccountAndLogin` API. In a nutshell, I was lazy to even try to figure out how it works with old API and where it should be added there, because it is going to be obsolete anyway and changes on desktop client side are required in that part of code (at least eventually). But yeah, we can do this both ways, both of which I personally don't like and would be happy to keep it at `CreateAccountAndLogin/RestoreAccountAndLogin`.

Reply:
@rasom if the only case when we need to have `"omit-transfers-history-scan?"` set to `true` is the same as the only case where we have `MnemonicRemoved` set to `false`, as stated in [this comment](https://github.com/status-im/status-desktop/pull/12483#discussion_r1377243312) why don't we use that fact on the status-go side? 

No need to do any changes on the desktop app side.